docs/reference/cluster/prevalidate-node-removal.asciidoc

```diff
@@ -0,0 +1,78 @@
+[[prevalidate-node-removal-api]]
+=== Prevalidate node removal API
+++++
+<titleabbrev>Prevalidate node removal</titleabbrev>
+++++
+
+This API is currently experimental for internal use by Elastic software only.
+
+NOTE: {cloud-only}
+
+Prevalidate node removal.
+
+[[prevalidate-node-removal-api-request]]
+==== {api-request-title}
+
+`GET /_internal/prevalidate_node_removal/<node_id>`
+
+[[prevalidate-node-removal-api-prereqs]]
+==== {api-prereq-title}
+
+* If the {es} {security-features} are enabled, you must have the `monitor` or `manage` <<privileges-list-cluster,cluster privilege>> to use this API.
+
+[[prevalidate-node-removal-api-desc]]
+==== {api-description-title}
+
+This API allows validating whether it is safe to remove the specified node(s) from the cluster. In case the cluster has a <<cluster-health,`red` cluster health status>>, this verifies that the removal of the node(s) would not risk removing the last remaining copy of an unassigned shard.
+
+The response includes the overall safety of the removal of the specified nodes, and a detailed response for each node. The node-specific part of the response also includes more details on why removal of that node might not be safe.
+
+The result of the prevalidation is expressed as either `YES`, `NO`, or `UNKNOWN`.
+
+[[prevalidate-node-removal-api-path-params]]
+==== {api-path-parms-title}
+
+`<node_id>`::
+    (Required, string) Comma-separated list of node IDs.
+
+[[prevalidate-node-removal-api-response-body]]
+==== {api-response-body-title}
+
+`overall_result`::
+    (object) Prevalidation result for the removal of all the provided nodes.
+
+`per_node_result`::
+    (object) Prevalidation result for the removal of each of the provided nodes.
+
+[[prevalidate-node-removal-api-example]]
+==== {api-examples-title}
+
+This example validates whether it is safe to remove the nodes `node1` and `node2`. The response indicates that it is safe to remove `node`, but it might not be safe to remove `node2`. Therefore, the overall prevalidation of the removal of the two nodes is `UNKNOWN`.
```

Comment:
```suggestion
This example validates whether it is safe to remove the nodes `node1` and `node2`. The response indicates that it is safe to remove `node1`, but it might not be safe to remove `node2`. Therefore, the overall prevalidation of the removal of the two nodes is `UNKNOWN`.
```